src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
+
+enum ItemsPerPage {
+  three = 3,
+  five = 5,
+  ten = 10,
+  twenty = 20,
+}
 
 export const App: React.FC = () => {
+  const [, setSearchParams] = useSearchParams();
+  const [perPage, setPerPage] = useState(ItemsPerPage.five);
+  const [currentPage, onPageChange] = useState(1);
+
+  const itemsOnPage = items.slice(
+    (currentPage - 1) * perPage,
+    (currentPage - 1) * perPage + perPage,
+  );
+
+  useEffect(() => {
+    setSearchParams({
+      page: currentPage.toString(),
+      perPage: perPage.toString(),
+    });
+  }, [currentPage, perPage, setSearchParams]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items{' '}
+        {itemsOnPage[0].split(' ')[1] && itemsOnPage[0].split(' ')[1]} -{' '}
+        {itemsOnPage[itemsOnPage.length - 1].split(' ')[1] &&
+          itemsOnPage[itemsOnPage.length - 1].split(' ')[1]}{' '}
```

Comment:
There's a potential issue with directly accessing array elements without checking if they exist. If `itemsOnPage` is empty, trying to access `itemsOnPage[0]` or `itemsOnPage[itemsOnPage.length - 1]` will result in an error. It's safer to perform a check before accessing these elements to avoid runtime errors.